tests/examples/authentication.test.js

```diff
@@ -15,18 +15,18 @@ class Session {
 
 class AuthenticatedSession {
   isAuthenticated = true;
-  content = MS.Object;
+  content = Object;
 
   logout() {
-    return this(AnonymousSession);
+    return this.set(AnonymousSession);
   }
 }
 
 class AnonymousSession {
   content = null;
   isAuthenticated = false;
-  authenticate(current, user) {
-    return this(AuthenticatedSession, { content: user });
+  authenticate(user) {
+    return this.set(AuthenticatedSession, { content: user });
```

Comment:
`set` becomes the instance analogue of `create` then?

I'm a little hesitant to use varargs since it means that the method can't be curried later on. 

One alterative would be to just call create directly:

```js
import { create } from 'microstates';
// .....
class Session {
  authenticate(username) {
    return create(AnonymousSession, { username });
  }
```

That has a nice simplicity and symmetry to it.

However, a `set` transition that allows changing the type _does_ open up some very interesting possibilities, because it means that you can type-shift down the tree:

```js
return this.user.address.set(PostOfficeBox, {number: 6542});
````

I'm not sure this is possible with bare `create`. Or maybe it is?

🤔